### PR TITLE
support data plane recording generate

### PR DESCRIPTION
**Pull request by jordan13game**

### CODE REVIEW

File: lib/apiScenario/gen/testRecordingApiScenarioGenerator.ts [hunk 1]
```diff
@@ -168,7 +171,7 @@ export class TestRecordingApiScenarioGenerator {
   ): Promise<RawScenario> {
     logger.info(`\nGenerating ${requestTracking.description}`);
     const testScenario: RawScenario = {
-      scenario: requestTracking.description,
+      scenario: requestTracking.description.replace(/\(/g, "_").replace(/\)/g, "_"),
```
Comment (leniatgh) [inline lib/apiScenario/gen/testRecordingApiScenarioGenerator.ts]:
```suggestion
      scenario: requestTracking.description.replace(/[\(\)]/g, "_"),
```
And any other special characters? like space,`!@#%^&*`?

File: lib/apiScenario/variableEnv.ts [hunk 2]
```diff
@@ -262,6 +262,31 @@ export class VariableEnv {
     }
   }
 
+  public tryResolveString(source: string): string {
+    const regex = variableRegex;
```
Comment (leniatgh) [inline lib/apiScenario/variableEnv.ts]: directly use `variableRegex` ?

Comment (leniatgh) [reply lib/apiScenario/variableEnv.ts]: what's the difference with `doResolveStringVariable`?

**Review by leniatgh — CHANGES_REQUESTED** [review]

File: lib/apiScenario/gen/testRecordingApiScenarioGenerator.ts (hunk 1 above)
Comment (jordan13game) [reply lib/apiScenario/gen/testRecordingApiScenarioGenerator.ts]: space may in filename, `!@#%^&*` is illegal in path

File: lib/apiScenario/variableEnv.ts (hunk 2 above)
Comment (jordan13game) [reply lib/apiScenario/variableEnv.ts]: This function uses `get` not `getRequired`, it won't throw error when it not found. And only resolve `string` type without `bool` and `int`

**Review by leniatgh — APPROVED** [review]
LGTM, please revert version change in package.json

**Pull request merged**